Script/Quest: Rivenwood Captives

Staleness89 opened the pull request:
**Changes proposed:**
 Adds the missing summons from Cocoons for the Howling Fjord quest Rivenwood Captives

**Target branch(es):** 3.3.5/master
both?
**Issues addressed:**
#18076

**Tests performed:** (Does it build, tested in-game, etc.)
Built and tested, not sure about the % on the spawns, guessed based on videos.
**Known issues and TODO list:**
The spawns don't react if the player is far away enough, figured a JUST_SUMMONED trigger in SmartAI to attack the summoner, which is technically the player, would be okay but I wanted to see if this okay as is first.

Staleness89 committed:
Script/Quest: Rivenwood Captives

Staleness89 commented:
I have failed. what did I do wrong?

DDuarte commented:
https://travis-ci.org/TrinityCore/TrinityCore/builds/168223782#L1115

DDuarte commented:
Also, change all tabs to 4 spaces and remove the hungarian notation (`uiRand` -> `rand` or similar).

jackpoz commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,67 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+				player->CastSpell(me, SPELL_SUMMON_FREED_MIST_WHISPER_SCOUT, true);
+				player->KilledMonsterCredit(NPC_MIST_WHISPER_SCOUT);
+			}
+			else if (uiRand >= 20)
```
is there a case where this line is reached and yet uiRand >= 20 evaluates to false ?

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,67 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+	SPELL_SUMMON_WINTERSKORN_ORACLE = 43285,
+	SPELL_SUMMON_FREED_MIST_WHISPER_SCOUT = 43289,
+	NPC_MIST_WHISPER_SCOUT = 24211
+};
```
try align those spells, please

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -595,4 +656,5 @@ void AddSC_howling_fjord()
     new npc_daegarn();
     new npc_mindless_abomination();
     new spell_mindless_abomination_explosion_fx_master();
+	new npc_riven_widow_cocoonAI();
```
codestyle

tkrokli replied:
Yes, it is caused by tabs instead of spaces
(github sets tab size = 8 spaces).

tkrokli replied:
Also, this is where the build fails. The `AddSC_howling_fjord()` line should be
```cpp
     new npc_riven_widow_cocoon();
```

Staleness89 replied:
duh.. I can't believe I missed that.

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,67 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+			uint8 uiRand = urand(0, 99);
+			
+			if (!player)
+				return;
```
you using tabs instead 4 spaces?
this "spaces" looks weird for me

Aokromes replied:
yes, also whitespace before if.

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,67 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+{
+	SPELL_SUMMON_BABY_RIVEN_WIDOWS, SPELL_SUMMON_DARKCLAW_BAT, SPELL_SUMMON_FANGGORE_WORG, SPELL_SUMMON_GJALERBRON_RUNECASTER, SPELL_SUMMON_GJALERBRON_SLEEPWATCHER, SPELL_SUMMON_GJALERBRON_WARRIOR,
+	SPELL_SUMMON_PUTRID_HORROR, SPELL_SUMMON_WINTERSKORN_BERSERKER, SPELL_SUMMON_WINTERSKORN_WOODSMAN, SPELL_SUMMON_WINTERSKORN_TRIBESMAN, SPELL_SUMMON_WINTERSKORN_ORACLE
+};
```
codestyle

Temppus commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,67 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+		void JustDied(Unit* killer) override
+		{
+			Player* player = killer->ToPlayer();
+			uint8 uiRand = urand(0, 99);
```
No need for variable or else if.

if (urand(0, 99) < 200
{
}
else
{
}

ariel- replied:
No need for urand either

```c++
if (roll_chance_i(20))
{
    // do something
}
else
{
    // do something
}

Staleness89 commented:
I will try to fix everything now, sorry this is my first pull request.

Staleness89 committed:
Fix coding style and build error

Staleness89 commented:
Hopefully that fixes everything. I will also make up an SQL file to fix the cocoon attacking in melee range, as well as SmartAI telling all the spawns to attack whoever destroys the cocoon.

Temppus commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,66 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+        void JustDied(Unit* killer) override
+        {
+            Player* player = killer->ToPlayer();
+            uint8 uiRand = urand(0, 99);
```
unused variable

Staleness89 committed:
Unused variable

Aokromes commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -656,5 +655,5 @@ void AddSC_howling_fjord()
     new npc_mindless_abomination();
     new spell_mindless_abomination_explosion_fx_master();
-	new npc_riven_widow_cocoonAI();
+	new npc_riven_widow_cocoon();
```
tab

Staleness89 committed:
Fix a tab

Staleness89 committed:
Is this what it wants?

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,65 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+{
+    SPELL_SUMMON_BABY_RIVEN_WIDOWS, SPELL_SUMMON_DARKCLAW_BAT, SPELL_SUMMON_FANGGORE_WORG, SPELL_SUMMON_GJALERBRON_RUNECASTER, SPELL_SUMMON_GJALERBRON_SLEEPWATCHER, SPELL_SUMMON_GJALERBRON_WARRIOR,
+    SPELL_SUMMON_PUTRID_HORROR, SPELL_SUMMON_WINTERSKORN_BERSERKER, SPELL_SUMMON_WINTERSKORN_WOODSMAN, SPELL_SUMMON_WINTERSKORN_TRIBESMAN, SPELL_SUMMON_WINTERSKORN_ORACLE
+};
```
code style suggested:

```
const uint32 rivenWidowCocoonVictims[11] =
 {
     SPELL_SUMMON_BABY_RIVEN_WIDOWS,
     SPELL_SUMMON_DARKCLAW_BAT, 
     SPELL_SUMMON_FANGGORE_WORG,
     SPELL_SUMMON_GJALERBRON_RUNECASTER,
     SPELL_SUMMON_GJALERBRON_SLEEPWATCHER,
     ...
 };

```

Keader commented:
Your last commit is wrong i think, i belive you can revert him.
And dont need any SQL querry for "link" c++ script for this creature? (i dont have acess for database right now for check it)

> I will try to fix everything now, sorry this is my first pull request.

Dont worry, the main objective of PRs is improved the code... And Trinity is a project for learning... so... dont worry xd

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -642,7 +642,7 @@ class npc_riven_widow_cocoon : public CreatureScript
 
     CreatureAI* GetAI(Creature* creature) const override
     {
-        return new npc_riven_widow_cocoon(creature);
```
```return new  npc_riven_widow_cocoonAI(creature);```

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -642,7 +642,7 @@ class npc_riven_widow_cocoon : public CreatureScript
     CreatureAI* GetAI(Creature* creature) const override
     {
-        return new npc_riven_widow_cocoon(creature);
+        return new npc_riven_widow_cocoon();
```
```return new  npc_riven_widow_cocoonAI(creature);```

Staleness89 committed:
This should do it

Staleness89 commented:
``` SQL
-- core script (and delete the incomplete SAI)
DELETE FROM `smart_scripts` WHERE `entry`=24210;
UPDATE `creature_template` SET `ScriptName`='npc_riven_widow_cocoon' WHERE `entry`= 24210;

-- set creatures that spawn to SmartAI (the ones that aren't already)
UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`= 24219;
UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`= 24073;
UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`= 24026;

-- Add an "attack summoner" on "just summoned" to each creature. Some of these creatures already have SmartAI, did all this to avoid errors.

DELETE FROM `smart_scripts` WHERE `entry`=24219;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(24219, 0, 0, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Baby Riven Widow  - On Just Summoned - Attack Summoner');

DELETE FROM `smart_scripts` WHERE `entry`=24073;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(24073, 0, 0, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Fearsome Horror  - On Just Summoned - Attack Summoner');

DELETE FROM `smart_scripts` WHERE `entry`=23661;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23661, 0, 0, 1, 8, 0, 100, 0, 43036, 0, 0, 0, 11, 43059, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Tribesman - On Spellhit \'Dismembering Corpse\' - Cast \'Gruesome, But Necessary: Vrykul On Dismember\'');
(23661, 0, 3, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Tribesman  - On Just Summoned - Attack Summoner');
(23661, 0, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 11, 43037, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Tribesman - On Aggro - Cast \'Gruesome, But Necessary: Kill Credit\'');
(23661, 0, 2, 0, 0, 0, 100, 0, 1000, 15000, 7200, 35000, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Tribesman - In Combat - Say Line 0');

DELETE FROM `smart_scripts` WHERE `entry`=23666;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23666, 0, 4, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Berserker  - On Just Summoned - Attack Summoner');
(23666, 0, 0, 1, 8, 0, 100, 0, 43036, 0, 0, 0, 11, 43059, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Berserker - On Spellhit \'Dismembering Corpse\' - Cast \'Gruesome, But Necessary: Vrykul On Dismember\'');
(23666, 0, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 11, 43037, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Berserker - On Spellhit \'Dismembering Corpse\' - Cast \'Gruesome, But Necessary: Kill Credit\'');
(23666, 0, 2, 0, 0, 0, 100, 0, 1000, 15000, 7200, 35000, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Berserker - In Combat - Say Line 0');
(23666, 0, 3, 0, 1, 0, 100, 0, 0, 60000, 15000, 70000, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Berserker - Out of Combat - Say Line 1');

DELETE FROM `smart_scripts` WHERE `entry`=23991;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23991, 0, 0, 0, 4, 0, 50, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Gjalerbron Warrior - On Agro - Say');
(23991, 0, 1, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Gjalerbron Warrior  - On Just Summoned - Attack Summoner');

DELETE FROM `smart_scripts` WHERE `entry`=23990;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23990, 0, 0, 0, 4, 0, 50, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Gjalerbron Rune-Caster - On Agro - Say');
(23990, 0, 1, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Gjalerbron Runecaster  - On Just Summoned - Attack Summoner');

DELETE FROM `smart_scripts` WHERE `entry`=23989;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23989, 0, 0, 0, 4, 0, 50, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Gjalerbron Sleep-Watcher - On Agro - Say');
(23989, 0, 1, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Gjalerbron SleepWatcher  - On Just Summoned - Attack Summoner');

DELETE FROM `smart_scripts` WHERE `entry`=23669;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23669, 0, 0, 0, 0, 0, 100, 0, 1000, 3500, 3000, 9000, 11, 43083, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Oracle - In Combat - Cast \'Frostbolt\'');
(23669, 0, 1, 2, 8, 0, 100, 0, 43036, 0, 0, 0, 11, 43059, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Oracle - On Spellhit \'Dismembering Corpse\' - Cast \'Gruesome, But Necessary: Vrykul On Dismember\'');
(23669, 0, 1, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Oracle  - On Just Summoned - Attack Summoner');
(23669, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 11, 43037, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Oracle - On Spellhit \'Dismembering Corpse\' - Cast \'Gruesome, But Necessary: Kill Credit\'');
(23669, 0, 3, 0, 0, 0, 100, 0, 1000, 15000, 7200, 35000, 5, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Oracle - In Combat - Say Line 0');

DELETE FROM `smart_scripts` WHERE `entry`=23959;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23959, 0, 0, 0, 38, 0, 100, 0, 0, 1, 0, 0, 69, 1, 0, 0, 0, 0, 0, 19, 24230, 50, 0, 0, 0, 0, 0, 'Darkclaw Bat - On Data Set 0 1 - Move To Closest Creature \'Feknut\'s Firecrackers Bunny\' (No Repeat) (Dungeon)');
(23959, 0, 1, 0, 34, 0, 100, 512, 0, 1, 0, 0, 45, 0, 1, 0, 0, 0, 0, 19, 24230, 5, 0, 0, 0, 0, 0, 'Darkclaw Bat - On Reached Point 1 - Set Data 0 1 (No Repeat) (Dungeon)');
(23959, 0, 2, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Darkclaw Bat  - On Just Summoned - Attack Summoner');

DELETE FROM `smart_scripts` WHERE `entry`=23662;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(23662, 0, 0, 0, 0, 0, 100, 0, 100, 1000, 5000, 14500, 11, 38557, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Woodsman - In Combat - Cast \'Throw\'');
(23662, 0, 1, 0, 0, 0, 100, 0, 1000, 4500, 4900, 8100, 11, 43410, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Woodsman - In Combat - Cast \'Chop\'');
(23662, 0, 2, 3, 8, 0, 100, 0, 43036, 0, 0, 0, 11, 43059, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Woodsman - On Spellhit \'Dismembering Corpse\' - Cast \'Gruesome, But Necessary: Vrykul On Dismember\'');
(23662, 0, 2, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Woodsman  - On Just Summoned - Attack Summoner');
(23662, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 11, 43037, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Woodsman - On Spellhit \'Dismembering Corpse\' - Cast \'Gruesome, But Necessary: Kill Credit\'');
(23662, 0, 4, 0, 0, 0, 100, 0, 1000, 15000, 7200, 35000, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Winterskorn Woodsman - In Combat - Say Line 0');

DELETE FROM `smart_scripts` WHERE `entry`=23662;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(24026, 0, 0, 0, 54, 0, 100, 512, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Fanggore Worg  - On Just Summoned - Attack Summoner');

-- fixes the cocoon being able to attack
UPDATE `creature_template` SET `unit_flags`=131076 WHERE `entry`=24210;
```

Keader commented:
@tkrokli  can you teach him to put this querrys in a sql file and push to PR?

Staleness89 commented:
Do I open a new issue and attach it there?

Staleness89 committed:
Script/Quest: Rivenwood Captives

Staleness89 commented:
hopefully I named it right

Keader commented:
sql file name should be something like: YYYY_MM_DD_world.sql
e.g.: 9955_22_52_world.sql
try some "value" that cant be reach (only for not give confflicts on merge). After merge some dev will rename this file.

Staleness89 committed:
Fix sql name

Aokromes commented:
I prefer 9999_99_99_18094_world.sql where 18094 is the PR id, on that way we can merge multiple PRs containing sql files on a row and use a single commit to rename all the merged files.

Staleness89 committed:
might as well while I'm here

Commit:
SQL improvements

- 1 update line is enough to set SmartAI for the 3 NPCs
- use only 1 DELETE FROM / INSERT INTO pair for the SAI lines
- added alignment spacing for readability in the SAI columns
- use double quotation marks (" ") around comments when single quotes are used within

Commit:
Remove copy-paste error from SQL file

Paste fail, inserted double DELETE FROM / INSERT INTO pair in previous edit

Keader commented on src/server/scripts/Northrend/zone_howling_fjord.cpp:
```diff
@@ -587,6 +587,74 @@ class spell_mindless_abomination_explosion_fx_master : public SpellScriptLoader
+        void Reset() override { }
+        void EnterCombat(Unit* /*who*/) override { }
+        void MoveInLineOfSight(Unit* /*who*/) override { }
+
```
When you dont override this functions, creatures obey the "Generic AI".
You just need override this functions with "nothing inside" when you realy need that this functions do nothing... Like if creature never should reset or never EnterCombat or never EnterInEvade, etc.

Aokromes commented:
what's missing here?

tkrokli commented:
@Staleness89 : https://github.com/TrinityCore/TrinityCore/issues/18076#issuecomment-255854746

- the cocoons also grant experience. They shouldn't.
```sql
UPDATE `creature_template` SET `flags_extra`=`flags_extra`|64 WHERE `entry`=24210;
```

Staleness89 committed:
Disable experience, add follow to Freed Winterhoof Longerunner's SAI and rename the file.

Staleness89 committed:
Merge branch '3.3.5' of https://github.com/Staleness89/TrinityCore into 3.3.5

Conflicts:
	sql/updates/world/3.3.5/9999_99_99_18094_world.sql

Staleness89 committed:
disable experience for cocoon, add follow on summon for freed winterhoof longrunner

Staleness89 committed:
disable experience for cocoon, add follow on summon for freed winterhoof longrunner

Staleness89 commented:
was having some trouble there, should be good to go now

Staleness89 committed:
fix syntax error

Staleness89 commented:
pull me plz

tkrokli commented:
Would like to, but I prefer that a TC member with more seniority does it, just to be on the safe side.

Aokromes committed:
Rename 9999_99_99_18094_world.sql to 2016_11_23_00_world.sql
